### DISCUSSION TITLE
Should Il2Cpp::perform be called more than once?

### DISCUSSION THREAD

**Discussion by Flechaa** [vfsfitvnm/frida-il2cpp-bridge#319] [Q&A] [answered]
In weird situations not wrapping specific code inside Il2Cpp::perform causes undefined behavior (like the frida script silently crashing) sometimes it could also lead to memory access violation but always seemingly random.

Sometimes it can happen while creating multiple objects and then calling methods (which most of the times don't use System objects).
I would like to know what would be the best practices when doing such things.

**Comment by vfsfitvnm** [comment, marked as answer]
These resources might be helpful: https://github.com/vfsfitvnm/frida-il2cpp-bridge/issues/239#issuecomment-1304831652, https://github.com/djkaty/Il2CppInspector/issues/155, https://github.com/vfsfitvnm/frida-il2cpp-bridge/discussions/296, https://github.com/vfsfitvnm/frida-il2cpp-bridge/discussions/302.

TL;DR: The best practice is to always wrap your code inside a `Il2Cpp::perform` block.
